cluster-api/pkg/controller/machineset/reconcile_test.go

```diff
@@ -205,15 +249,49 @@ func machineFromMachineSet(machineSet *v1alpha1.MachineSet, name string) *v1alph
 		Spec:       machineSet.Spec.Template.Spec,
 	}
 
-	amachine.Name = name
-	amachine.GenerateName = fmt.Sprintf("%s-", machineSet.Name)
+	m.Name = name
+	m.GenerateName = fmt.Sprintf("%s-", machineSet.Name)
+
+	return m
+}
+
+func machineFromMachineSet(machineSet *v1alpha1.MachineSet, name string) *v1alpha1.Machine {
+	m := machineWithoutOwnerRefs(machineSet, name)
+	m.ObjectMeta.OwnerReferences = []metav1.OwnerReference{*metav1.NewControllerRef(machineSet, controllerKind)}
+	return m
+}
+
+func machineWithPreservedExistingOwnerRefs(machineSet *v1alpha1.MachineSet, name string) *v1alpha1.Machine {
```

Comment:
(nit)

Took me a while to figure out what this method was up to. Perhaps something like "machineWithManyOwnerRefs" might be more clear? or perhaps have a method like "AddOwnerRef" that tacks on an additional owner ref to OwnerReferences?

Reply:
renamed